File: 백준/Silver/18111. 마인크래프트/마인크래프트.py
from sys import stdin

input = stdin.readline

n, m, b = list(map(int, input().rsplit()))
ground = [list(map(int, input().rsplit())) for _ in range(n)]

min_time = int(1e9)
max_height = 0

for height in range(257):
    time_cost = 0
    cur_block = b

    for r in range(n):
        for c in range(m):
            if ground[r][c] > height:
                time_cost += 2 * (ground[r][c] - height)
                cur_block += ground[r][c] - height
            else:
                time_cost += height - ground[r][c]
                cur_block -= height - ground[r][c]

    if cur_block < 0:
        continue

    if min_time >= time_cost:
        min_time = time_cost
        max_height = height

print(min_time, max_height)

File: 백준/Silver/18111. 마인크래프트/마인크래프트.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    static int n;
    static int m;
    static int b;
    static int[][] ground;
    static BufferedReader br;
    static StringTokenizer st;


    public static void main(String[] args) throws Exception {
        br = new BufferedReader(new InputStreamReader(System.in));

        st = new StringTokenizer(br.readLine());
        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        b = Integer.parseInt(st.nextToken());

        ground = new int[n][m];
        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            for (int j = 0; j < m; j++) {
                ground[i][j] = Integer.parseInt(st.nextToken());
            }
        }

        int minTime = Integer.MAX_VALUE;
        int maxHeight = 0;

        for (int height = 0; height < 257; height++) {
            int timeCost = 0;
            int curBlock = b;

            for (int r = 0; r < n; r++) {
                for (int c = 0; c < m; c++) {
                    if (ground[r][c] > height) {
                        timeCost += 2 * (ground[r][c] - height);
                        curBlock += ground[r][c] - height;
                    } else {
                        timeCost += height - ground[r][c];
                        curBlock -= height - ground[r][c];
                    }
                }
            }

            if (curBlock < 0) {
                continue;
            }

            if (minTime >= timeCost) {
                minTime = timeCost;
                maxHeight = height;
            }

        }
        
        System.out.println(minTime + " " + maxHeight);
    }
}


